### DISCUSSION TITLE
question about pls regression

### DISCUSSION THREAD

**Discussion by onethursday** [spectrochempy/spectrochempy#493] [Q&A] [answered]
Thanks for your wonderful work! This is the best chemometric toolbox for python.
I want to use it for pls regression, but I dont know how to do it? Can you share a example or some code.

**Comment by atravert** [comment]
Thank you for your kind message. SCPy is not the best (yet ;-) ), because PLS is not implemented, but this is in our plans.
For PLS regresssion, I suugest you using SCPy for preprocessing your spectra, the extract the NDDataset.data fields (ndarrays) and use [scikit-learn PLS](https://scikit-learn.org/stable/modules/generated/sklearn.cross_decomposition.PLSRegression.html#sklearn-cross-decomposition-plsregression). 
Best,

**Comment by atravert** [comment]
Hello @onethursday,

I hope you are doing well. FYI we have just added the PLS regression in the dev version of SCPy, see:

example: https://www.spectrochempy.fr/latest/gallery/auto_examples/analysis/plot_pls.html
and doc: https://www.spectrochempy.fr/latest/reference/generated/spectrochempy.PLSRegression.html

Please let us know if you have a chance to test it opn your data: your feedback would be valuable tu us !

Best regards
Arnaud

**Comment by atravert** [comment, marked as answer]
Hello @onethursday,

The PLS regression is now in the last stable version of SCPy, see:

example: https://www.spectrochempy.fr/stable/gallery/auto_examples/analysis/plot_pls.html
and doc: https://www.spectrochempy.fr/stable/reference/generated/spectrochempy.PLSRegression.html

Let us know if you have a chance to test it or any questions: your feedback is valuable to us.

Best regards
Arnaud
